apps/site/components/Downloads/DownloadReleasesTable/index.tsx

```diff
@@ -32,25 +39,37 @@ const DownloadReleasesTable: FC = () => {
       </thead>
       <tbody>
         {releaseData.map(release => (
-          <tr key={release.major}>
-            <td data-label="Version">v{release.major}</td>
-            <td data-label="LTS">{release.codename || '-'}</td>
-            <td data-label="Date">
-              <FormattedTime date={release.currentStart} />
-            </td>
-            <td data-label="Date">
-              <FormattedTime date={release.releaseDate} />
-            </td>
-            <td data-label="Status">
-              <Badge kind={BADGE_KIND_MAP[release.status]} size="small">
-                {release.status}
-                {release.status === 'End-of-life' ? ' (EoL)' : ''}
-              </Badge>
-            </td>
-            <td className="download-table-last">
-              <DetailsButton versionData={release} />
-            </td>
-          </tr>
+          <>
+            <tr key={release.major}>
+              <td data-label="Version">v{release.major}</td>
+              <td data-label="LTS">{release.codename || '-'}</td>
+              <td data-label="Date">
+                <FormattedTime date={release.currentStart} />
+              </td>
+              <td data-label="Date">
+                <FormattedTime date={release.releaseDate} />
+              </td>
+              <td data-label="Status">
+                <Badge kind={BADGE_KIND_MAP[release.status]} size="small">
+                  {release.status}
+                  {release.status === 'End-of-life' ? ' (EoL)' : ''}
+                </Badge>
+              </td>
+              <td>
+                <LinkWithArrow
+                  className="cursor-pointer"
+                  onClick={() => setCurrentModal(release.version)}
+                >
+                  {t('components.downloadReleasesTable.details')}
+                </LinkWithArrow>
+              </td>
+            </tr>
+            <ReleaseModal
+              release={release}
+              open={currentModal === release.version}
+              onOpenChange={open => open || setCurrentModal(undefined)}
```

Comment:
Maybe the modalProps could just be `onClose` since we know when it is open or not ;)

So tha `onClose` becomes just a `fn`